src/pluggy/_callers.py

```diff
@@ -20,16 +19,38 @@
 
 # Need to distinguish between old- and new-style hook wrappers.
 # Wrapping with a tuple is the fastest type-safe way I found to do it.
-Teardown = Union[
-    tuple[Generator[None, Result[object], None], HookImpl],
-    Generator[None, object, object],
-]
+Teardown = Generator[None, object, object]
+
+
+def run_legacy_hookwrapper(
```

Comment:
We call this "old-style" in the docs, perhaps:

```suggestion
def run_old_style_hookwrapper(
```